### PR TITLE
added doPost function

### PR DESCRIPTION
**Pull request by claraeverett**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -27,12 +27,15 @@
...
   @Override
-  public void init(){
-      messages = new ArrayList<>();
-      messages.add("Bonjour");
-      messages.add("Ciao");
-      messages.add("Hola");  
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    messages = new ArrayList<>();
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: If you create the array here, you get will return only 1 comment. You can create it where it is declared and it will accumulate the messages.
private List<String> messages = new ArrayList<>()

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -27,12 +27,15 @@
...
+    messages.add(text);
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: You can change this to messages.add(request.getParameter("text-input")). You don't need a new variable, since you are just getting it from the form and adding it to messages.

If you prefer to have the new variable, move the declaration closer just above this line, so that it is declared close to where it is used.

So 
String text = request.getParameter("text-input");
messages.add(text);

File: portfolio/src/main/webapp/index.html [hunk 3]
```diff
@@ -66,6 +66,14 @@ <h1>Random Quotes</h1>
       <div id="quote-container"></div>
 </div>
 
+<form action="/data" method="POST">
+      <p>Enter a comment here:</p>
+      <textarea name="text-input"></textarea>
+      <br/><br/>
+
+      <input type="submit" />
```
Comment (sbahaud) [inline portfolio/src/main/webapp/index.html]: The formatting seems off. Use 2 space indent.
<form ...>
  <p> ...
  etc

Comment (claraeverett) [reply portfolio/src/main/webapp/index.html]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (claraeverett) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (claraeverett) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -26,13 +29,21 @@
...
+    response.getWriter().println(text);
+    Entity commentEntity = new Entity("Task");
+    commentEntity.setProperty("comment", text);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+    
+    response.sendRedirect("/index.html");
   }
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Looks like you added datastore changes to the same PR. Would it be possible to move to a separate PR, so that I can review it separateley.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -26,13 +29,21 @@
...
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("text-input");
+    messages.add(text);
+    response.getWriter().println(text);
+    Entity commentEntity = new Entity("Task");
+    commentEntity.setProperty("comment", text);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Can you move the datastore entity creation and write to a separate function? It make the code more reaadable

Something like, writeCommentToDataStore(text);

File: portfolio/src/main/webapp/index.html [hunk 6]
```diff
@@ -66,6 +66,13 @@ <h1>Random Quotes</h1>
       <div id="quote-container"></div>
 </div>
 
+<form action="/data" method="POST">
+      <p>Enter a comment here:</p>
+        <textarea name="text-input"></textarea>
+        <br/><br/>
+        <input type="submit" />
+</form>
```
Comment (sbahaud) [inline portfolio/src/main/webapp/index.html]: Please fix the formatting here. 2 space indent.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 5 above)
Comment (claraeverett) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/webapp/index.html (hunk 6 above)
Comment (claraeverett) [reply portfolio/src/main/webapp/index.html]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 4 above)
Comment (claraeverett) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Done

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -26,15 +32,41 @@
...
+    Query query = new Query("title");
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<String> tasks = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String title = (String) entity.getProperty("title");
+
+      tasks.add(title);
+    }
```
Comment (sbahaud) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: I would update the doGet method to do the read, instead of updating the doPost. Also can you move the get changes to a separate PR. If more code is added to the PR after I review it, it is hard for me track the changes.
